content/learning-paths/smartphones-and-mobile/using-neon-intrinsics-to-optimize-unity-on-android/2-arm-neon.md

```diff
@@ -0,0 +1,66 @@
+
+## Arm Neon SIMD
+
+[Arm Neon](https://developer.arm.com/Architectures/Neon) is an advanced “SIMD” extension to the Arm architecture instruction set.
+
+SIMD stands for Single Instruction Multiple Data. It is a paradigm that refers to processing multiple data items using a single native CPU instruction. The process of converting code such that it uses specific SIMD instructions to process more data in parallel is called vectorization. If the conversion is done automatically by a compiler, it is called auto-vectorization.
+
+If done manually, developers are usually free to choose to write their SIMD code directly in the native CPU assembly language, or they can choose to use function-like statements that are known to the compiler. These function-like statements are called intrinsics. The compiler is able to translate intrinsics to native CPU instructions. Used correctly they can increase the performance of your code.
+
+## How does Arm Neon work?
+
+Each Neon instruction processes multiple data items at once by using vector registers that are split into “lanes”. The number of lanes that fit into a vector register varies according to the size of the data that you want to use.
+
+For example, applications compiled for the AArch64 architecture (64-bit Arm microprocessors) have access to 32 vector registers with 128-bits each. That is 16 bytes per register.
+
+![Vector registers#center](images/vector-registers.png)
+
+Each 128-bit register (Vn), depicted in the above diagram, shows how a vector register can be used to hold:
+
+- 2 64-bit (or double precision) floating point or integer values
+
+- 4 32-bit (or single precision) floating point or integer values
+
+- 8 16-bit (or half precision) floating point or integer values
+
+- 16 8-bit integer values
+
+## How do we use Neon intrinsics in Unity?
+Neon intrinsics have traditionally been used from C and C++ but by using the Unity Burst compiler you can leverage Neon intrinsics directly in your own C# scripts.
+
+Developers can use Arm Neon in multiple ways.
+
+- Import Neon-enabled libraries
+
+- Burst compiler auto-vectorization
+
+- Handwritten Neon intrinsics
+
+You can also use a mixture of these methods should you need to.
+
+### Using Neon by importing libraries
+If you’d like to try using a Neon-supported library, an easy start would be the [Unity.Mathematics](https://docs.unity3d.com/Packages/com.unity.mathematics@1.3/manual/index.html) package from Unity.
+
+We will first look at auto-vectorization by the Burst compiler and then look at optimizing further by hand-writing Neon intrinsics. In both cases you can provide hints to the compiler to help it auto-vectorize. You can follow [best practice](#best-practice) to organize your code and data to make them optimal for Neon.
+
+### Using Neon through Burst auto-vectorization
```

Comment:
do you mean to move the "we will first look at.." paragraph? if so i think you're right - it will be better at the end of the "how do we use neon intrinsics in unity" section above it

Reply:
I was meaning line 57 could be above line 55, but other re-orderings are possible. Just headings felt a little out of sync with content.

Reply:
i moved the line 55 text down and created "next step" - i think it closes the section better.
done